app/src/main/java/com/github/se/cyrcle/ui/add/location/LocationPicker.kt

```diff
@@ -0,0 +1,80 @@
+package com.github.se.cyrcle.ui.add.location
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import com.github.se.cyrcle.model.map.MapViewModel
+import com.github.se.cyrcle.model.map.MapViewModel.LocationPickerState
+import com.github.se.cyrcle.model.parking.Location
+import com.github.se.cyrcle.ui.add.location.overlay.Crosshair
+import com.github.se.cyrcle.ui.add.location.overlay.RectangleSelection
+import com.github.se.cyrcle.ui.map.MapConfig
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.navigation.Screen
+import com.mapbox.maps.MapView
+import com.mapbox.maps.extension.compose.DisposableMapEffect
+import com.mapbox.maps.extension.compose.MapboxMap
+import com.mapbox.maps.plugin.gestures.gestures
+
+@Composable
+fun LocationPicker(
+    navigationActions: NavigationActions,
+    mapViewModel: MapViewModel = MapViewModel()
+) {
+  LaunchedEffect(Unit) {
+    mapViewModel.updateLocation(null)
+    mapViewModel.updateLocationPickerState(LocationPickerState.NONE_SET)
+  }
+  val mapViewportState = MapConfig.createMapViewPortStateFromViewModel(mapViewModel)
+  val mapView = remember { mutableStateOf<MapView?>(null) }
+  val locationPickerState by mapViewModel.locationPickerState.collectAsState()
+  Scaffold(
+      bottomBar = { LocationPickerBottomBar(navigationActions, mapViewModel, mapView) },
+      topBar = { LocationPickerTopBar(mapViewModel) }) { padding ->
+        MapboxMap(
+            mapViewportState = mapViewportState,
+            style = { MapConfig.DefaultStyle() },
+            modifier = Modifier.testTag("LocationPickerScreen").padding(padding),
+        ) {
+          DisposableMapEffect { mapViewInstance ->
+            mapView.value = mapViewInstance
+            onDispose {}
+          }
+        }
+        RectangleSelection(mapViewModel, padding)
+        Crosshair(mapViewModel, padding)
+      }
+
+  LaunchedEffect(locationPickerState) {
```

Comment:
I'm not sure it's extremely useful to disable the gestures on the map. I think the rectangle should take care of capturing the inputs?

Reply:
Oh yeah my bad this is some old code from a previous implementation, I have to remove it